src/Server/AwsCloudFormationTemplates.fs

```diff
@@ -12,481 +9,221 @@ open Shared
-    let resourceType = spec.pulumiType
-    let logicalId = resource.logicalId.Replace("-", "_")
-    {
-        resourceType = resourceType
-        logicalId = logicalId
-        importId = importId
-    }
-
-let validateAppScalingPolicy
-    (resource: AwsCloudFormationResource)
-    (resourceData: Dictionary<string, Dictionary<string,string>>)
-    (spec: CustomRemapSpecification) 
-    : bool =
-    (resourceData.ContainsKey resource.logicalId)
-    && (hasImportIdentityParts resource resourceData spec)
-    && ((((resourceData[resource.logicalId])["ScalingTargetId"]).Split("|")).Length = 3)
-
-let remapAppScalingPolicy
-    (resource: AwsCloudFormationResource)
-    (resourceData: Dictionary<string, Dictionary<string,string>>)
-    (resourceContext: AwsResourceContext)
-    (spec: CustomRemapSpecification) 
-    : RemappedSpecResult = 
-    let data = resourceData[resource.logicalId]
-    let scalingTargetIdParts = data["ScalingTargetId"].Split("|")
-    let importId = 
-        if scalingTargetIdParts.Length = 3 then
-            [scalingTargetIdParts[2]; scalingTargetIdParts[0]; scalingTargetIdParts[1]; data["PolicyName"]]
-            |> String.concat spec.delimiter
-        else ""
-                
-    let resourceType = spec.pulumiType
-    let logicalId = resource.logicalId.Replace("-", "_")
-    {
-        resourceType = resourceType
-        logicalId = logicalId
-        importId = importId
-    }
-
-let remapAppScalableTarget
-    (resource: AwsCloudFormationResource)
-    (resourceData: Dictionary<string, Dictionary<string,string>>)
-    (resourceContext: AwsResourceContext)
-    (spec: CustomRemapSpecification) 
-    : RemappedSpecResult = 
-    let data = resourceData[resource.logicalId]
-    let physicalIdParts = data["Id"].Split("|")
-    let importId = 
-        if physicalIdParts.Length = 3 then
-            [physicalIdParts[2]; physicalIdParts[0]; physicalIdParts[1]]
-            |> String.concat spec.delimiter
-        else ""
-                
-    let resourceType = spec.pulumiType
-    let logicalId = resource.logicalId.Replace("-", "_")
-    {
-        resourceType = resourceType
-        logicalId = logicalId
-        importId = importId
-    }
-
-let validateSecurityGroupRule
-    (resource: AwsCloudFormationResource)
-    (resourceData: Dictionary<string, Dictionary<string,string>>)
-    (spec: CustomRemapSpecification) 
-    : bool =
-    ((resourceData[resource.logicalId]).ContainsKey "IpProtocol") &&
-    ((resourceData[resource.logicalId]).ContainsKey "GroupId")
-
-let filterSecurityGroupRules
+let tryFindSecurityGroupRule
     (data: Dictionary<string,string>)
-    (rules: Map<string,JObject>)
-    : Map<string,JObject> =
+    (rules: Map<string,JObject>) =
     rules
     |> Map.filter (fun id props -> 
         props.Property("GroupId").Value.ToString() = data["GroupId"])
-    |> Map.filter (fun id props ->
+    |> Map.tryFindKey (fun id props ->
```

Comment:
potential issue here is that the previous code wouldn't match a rule unless it was the only one that matched - looks like this function will take the first match instead - I'd lean more conservative on that since we wouldn't want to import the wrong rule

Reply:
I see what you mean. Changed it to:
```fs
Map.filter (...)
|> Map.toList
|> function
    // only when one rule is matched, we return they key being the import ID
    | [ securityGroupImportId, _ ] -> Some securityGroupImportId
    | _ -> None
```